_posts/2015-04-09-flexibility-when-releasing-a-new-product-peace-corps-new-donation-platform.md

```diff
@@ -0,0 +1,172 @@
+
+Progressive Enhancement
+-----------------------
+
+Our effort to refresh the Peace Corps’ donations platform stemmed from
+many goals, such as highlighting volunteer stories (to humanize the
+agency,) creating an experience optimized for mobile devices, and
+simplifying the donation process. The last point proved particularly
+challenging, as we had little control over the payment processor and
+minimal flexibility in the collections form. Luckily, we *could* reduce
+the number of steps between landing on the site and donating. When the
+user found a compelling story, they could select a donation amount
+in-line, turning what used to be a four-step process into a single
+click.
+
+![screenshot: workflow of donation pathways on Peace Corps site](/assets/blog/peacecorps/image03.png)
+
+Our ideal interface would have been a seamless user experience, but once
+scoped would have required a heavy lift from our front end and
+accessibility teams. We would need to dynamically generate the interface
+and provide a fallback for users without scripting support. With an eye
+on the clock, we chose instead to create a simpler form using radio
+selectors to allow users to select a donation amount. Though this felt
+clunky and defeated our single-click goal, it worked, so that’s where we
+started. The budget eventually allowed for additional improvements to
+this interface to be prioritized. Currently, the experience for most
+users is roughly one step away from our initial design, but it *remains*
+a step away. However, since we released without our optimal solution, we
+bought time to build some other features (perhaps skipping payment
+confirmation, perhaps additional sort options.) Stepping back and
+re-prioritizing allowed us to make the product better on the whole.
+
+In another situation, starting with a simple solution ultimately led us
+down a worse path. Our most complex user interaction offers multiple
+filters and categorization mechanisms for selecting how a user’s
+donation will be used by the Peace Corps Volunteers. A reasonable
+technical architecture for this sort of page would involve asynchronous
+requests, where data is transmitted in chunks instead of in one large
+block, so that the browser only loads what is needed. We chose to start
+with a simpler approach, however, by including all content in the
+initial page load and simply hiding elements based on filters. With
+either strategy, users selecting a specific country would only see
+projects and funds relevant to that country, but the latter approach
+requires much more markup. Not long before launching, we realized that
+this approach led to massive page sizes when using live data. We quickly
+hacked around this problem via compression on our caching layer, and we
+later added additional patches to minimize redundant markup. We’re still
+not where we should be with regards to site load time because our
+initial approach was too simplistic.
+
+Zero Downtime
```

Comment:
Suggested title for this section: "Zero Downtime and Caching Overkill" or "Caching Overkill".